tcping.go

```diff
@@ -385,46 +386,43 @@ func findMinAvgMaxRttTime(timeArr []float32) rttResults {
 	return rttResults
 }
 
-/* Calculate cumulative time */
-func calcTime(time uint) calculatedTimeString {
-	var timeStr string
-
-	hours := time / (60 * 60)
-	timeMod := time % (60 * 60)
-	minutes := timeMod / (60)
-	seconds := timeMod % (60)
-
-	/* Calculate hours */
-	if hours >= 2 {
-		timeStr = fmt.Sprintf("%d hours %d minutes %d seconds", hours, minutes, seconds)
-		return timeStr
-	} else if hours == 1 && minutes == 0 && seconds == 0 {
-		timeStr = fmt.Sprintf("%d hour", hours)
-		return timeStr
-	} else if hours == 1 {
-		timeStr = fmt.Sprintf("%d hour %d minutes %d seconds", hours, minutes, seconds)
-		return timeStr
-	}
-
-	/* Calculate minutes */
-	if minutes >= 2 {
-		timeStr = fmt.Sprintf("%d minutes %d seconds", minutes, seconds)
-		return timeStr
-	} else if minutes == 1 && seconds == 0 {
-		timeStr = fmt.Sprintf("%d minute", minutes)
-		return timeStr
-	} else if minutes == 1 {
-		timeStr = fmt.Sprintf("%d minute %d seconds", minutes, seconds)
-		return timeStr
-	}
-
-	/* Calculate seconds */
-	if seconds >= 2 {
-		timeStr = fmt.Sprintf("%d seconds", seconds)
-		return timeStr
-	} else {
-		timeStr = fmt.Sprintf("%d second", seconds)
-		return timeStr
+// calcTime creates a human-readable string for a given duration
+func calcTime(duration time.Duration) calculatedTimeString {
+	hours := math.Floor(duration.Hours())
+	if hours > 0 {
+		duration -= time.Duration(hours * float64(time.Hour))
+	}
+
+	minutes := math.Floor(duration.Minutes())
+	if minutes > 0 {
+		duration -= time.Duration(minutes * float64(time.Minute))
+	}
+
+	seconds := duration.Seconds()
```

Comment:
Probably the ugliest part of the entire PR, couldn't come up with anything better

Reply:
Hey @ravsii,

Thanks a lot for your wonderful contributions!

On the first look, looks good to me.

Tomorrow, I will finalize the review of your other PR and then get to this one.

p.s. don't worry about "ugliness". There are a lotta code smells already existing that I need to fix if time lets me.